test/e2e-common.sh

```diff
@@ -27,8 +27,7 @@ source $(dirname $0)/../vendor/github.com/tektoncd/plumbing/scripts/e2e-tests.sh
 # from our tests withouth having to go to an external registry.
 function add_sidecar_registry() {
     cp ${1} ${TMPF}.read
-
-    cat ${TMPF}.read | python3 -c 'import yaml;f=open(0, encoding="utf-8"); data=yaml.load(f.read(), Loader=yaml.FullLoader);data["spec"]["sidecars"]=[{"image":"registry", "name": "registry"}];print(yaml.dump(data, default_flow_style=False));' > ${TMPF}
+    cat ${TMPF}.read | python3 -c 'import yaml;f=open(0, encoding="utf-8"); data=yaml.load(f.read(), Loader=yaml.FullLoader);sidecars=data["spec"].get("sidecars"); sidecars = [[], data["spec"].get("sidecars")][sidecars != None]; sidecars.append({"image":"registry", "name": "registry"}); data["spec"]["sidecars"]=sidecars;print(yaml.dump(data, default_flow_style=False));' > ${TMPF}
```

Comment:
This is frankly indecipherable. Can we make this more readable somehow?

Reply:
yes we can.. let's maybe add this fix and work on refactoring this?

Reply:
I 110% percent agree. There is tool already designed to do this kind of substitution called yq. I don't know why there is an arbitrary line of python in the first place. However, I feel like this someone else has had to say something before. This had to be put there for some reason. idk, why tho...